src/dotnet-scaffolding/Microsoft.DotNet.Scaffolding.Core/Builder/ScaffoldRunnerBuilder.cs

```diff
@@ -72,13 +74,41 @@ private void AddDefaultServices()
 
     private void AddCoreServices()
     {
-        Services.AddLogging();
-        Logging.AddCleanConsoleFormatter();
+        //get some logging properties from environment variables
+        var isVerboseEnabled = Environment.GetEnvironmentVariable(ScaffolderConstants.ENABLE_VERBOSE_LOGGING)?.Equals("true", StringComparison.OrdinalIgnoreCase) == true;
+        var isLogToFileEnabled = Environment.GetEnvironmentVariable(ScaffolderConstants.LOG_TO_FILE)?.Equals("true", StringComparison.OrdinalIgnoreCase) == true; ;
+        // Configure Serilog Logger
+        var loggerConfig = new LoggerConfiguration().WriteTo.Sink(new AnsiConsoleSink());
+        loggerConfig.MinimumLevel.Information();
+        if(isVerboseEnabled)
+        {
+            loggerConfig.MinimumLevel.Verbose();
+        }
+
+        if (isLogToFileEnabled)
+        {
+            var currentDirectory = Directory.GetCurrentDirectory();
+            if (!string.IsNullOrEmpty(currentDirectory))
+            {
+                var logPath = Path.Combine(currentDirectory, _defaultLogFolder, "dotnet-scaffold.txt");
+                loggerConfig.WriteTo.File(logPath, rollingInterval: RollingInterval.Minute);
```

Comment:
Does RollingInterval.Minute mean you get a new log file every minute? If so, is that by wall clock or runtime? Like if you start running the scaffolder at 1:00:55 and it runs for 10 seconds to 1:01:05, do you end up with two log files for the run? Or just one?

Reply:
testing all these scenarios out today, will follow up soon.

Reply:
yes, so we get 2 log files on the minute switch, and multiple executions do not create duplicate files, it adds to the same file. adding more info in the description. I think I prefer logs every minute vs hourly. Any thoughts on this?

Reply:
I think the most common scenario we're going to want these log files from is when we ask customers "send us the logs from your most recent run". From my perspective, it would be nice if that was one file that contained only the logs from that current run (or, alternatively, one log file per scaffolder or similar distinction). So we don't have to ask if it was the run at 4am or the run at 4:30am (or 4:00:00am vs 4:00:30am for minute rolling), and we don't have to ask to make sure they send all the files because it might have gotten split across multiple and then we're left piecing them back together.

I see the rolling interval from serilog being more interesting for log files that are being regularly generated - a web app that's running all the time and it dumps logs every few seconds all day or whatever. The scaffolder runs are discrete runs that shouldn't last more than a couple minutes at most, though it'd be interesting to have all of those minutes together in one file.

I wonder if we should consider a different approach of just picking a filename based on the execution time of the tool and not rolling it. Whatever goes into that log file goes into it. So we get PATH\dotnet-scaffold_20240612125152.log for that entire run. 

Thoughts?

Reply:
yeah that makes a lot of sense, one file per execution/run. also do we prefer .log extension over .txt?

Reply:
nvm, .NET seems to use .log convention so will do that

Reply:
addressed, updating the PR description